src/services/notification.ts

```diff
@@ -16,55 +16,170 @@
  */
 
 import Receiver = threema.Receiver;
+import SettingsService = threema.SettingsService;
 export class NotificationService implements threema.NotificationService {
 
+    private static SETTINGS_NOTIFICATIONS = 'notifications';
+    private static SETTINGS_NOTIFICATION_PREVIEW = 'notificationPreview';
+
     private $log: ng.ILogService;
     private $window: ng.IWindowService;
     private $state: ng.ui.IStateService;
 
+    private settingsService: SettingsService;
+    private logTag = '[Notification Service]';
+
     // Whether user has granted notification permission
-    private mayNotify: boolean = null;
+    private notificationPermission: boolean = null;
+    // Whether user wants to receive desktop notifications
+    private desktopNotifications: boolean = null;
+    // Whether the browser supports notifications
+    private notificationAPIAvailable: boolean = null;
+    // Whether the user wants notification preview
+    private notificationPreview: boolean = null;
 
     // Cache notifications
     private notificationCache: any = {};
 
-    public static $inject = ['$log', '$window', '$state'];
-    constructor($log: ng.ILogService, $window: ng.IWindowService, $state: ng.ui.IStateService) {
+    public static $inject = ['$log', '$window', '$state', 'SettingsService'];
+
+    constructor($log: ng.ILogService, $window: ng.IWindowService,
+                $state: ng.ui.IStateService, settingsService: SettingsService) {
         this.$log = $log;
         this.$window = $window;
         this.$state = $state;
+        this.settingsService = settingsService;
     }
 
     /**
      * Ask user for desktop notification permissions.
      *
      * Updates internal `maxNotify` flag. If the user accepts, the is set to to
      * true. If the user declines, the flag is set to false. If the
-     * Notifications API is not available, the flag is set to null.
+     * user makes no choice, the flag is set to null
```

Comment:
Can you update the comment with the correct flag variables? `mayNotify` (there was a typo in there) doesn't exist anymore.